### PR TITLE
DBZ-7062 Update transformation property "delete.tombstone.handling.mode" to debezium doc

### PR DESCRIPTION
**Pull request by harveyyue**
https://issues.redhat.com/browse/DBZ-7062

### CODE REVIEW

**Comment by harveyyue** [comment]
Will update the new property `delete.tombstone.handling.mode` to doc

**Comment by harveyyue** [comment]
@rk3rn3r Could you please review these doc change again?😊

**Comment by jpechane** [comment]
@roldanbob Could you please take a look at it too?

@harveyyue I am a bit confused about the additional code changes. Why is that? It seems to be unrelated to docs changes. If it is a bugfix could you please create additional Jira and create separate commit for the bugfix? Thanks

**Comment by harveyyue** [comment]
@jpechane OK, will create a separate commit for this hotfix.

File: documentation/modules/ROOT/pages/transformations/event-flattening.adoc [hunk 1]
```diff
@@ -268,6 +271,24 @@ Specify `none` to keep the change event record in the stream. The record contain
...
+
+`drop`:: The SMT removes both the `DELETE` event and `TOMBSTONE` from the stream.
+`tombstone`:: The SMT only keeps the `TOMBSTONE` in the stream. The record contains only `"value": "null"`.
+`rewrite`:: The SMT keeps the change event record in the stream and edit the record to have a `value` field that contains the key/value pairs that were in the `before` field and also add `+__deleted: true+` to the `value` and remove the `TOMBSTONE`.
+This is another way to indicate that the record has been deleted.
+`rewrite-with-tombstone`:: The SMT has the same behavior from `rewrite` except to keep the `TOMBSTONE`.
+[NOTE]
+====
+Note: When specify a value from above will enable this new config in event flatting SMT, it has a higher priority than the deprecated xref:extract-new-record-state-drop-tombstones[`drop.tombstones`] and xref:extract-new-record-state-delete-handling-mode[`delete.handling.mode`].
+====
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/event-flattening.adoc:291]:
```suggestion
`drop`:: The SMT removes both `DELETE` events and `TOMBSTONE`records from the stream.
`tombstone` (default):: The SMT retains `TOMBSTONE` records in the stream.
The `TOMBSTONE` record contains only the following value: `"value": "null"`.
`rewrite`:: The SMT retains the change event record in the stream and makes the following changes:

* Adds a `value` field to the record that contains the key/value pairs from the `before` field of the original record.
* Adds `+__deleted: true+` to the `value` of the record.
* Removes `TOMBSTONE` records.
+
This setting provides another way to indicate that the record has been deleted.

`rewrite-with-tombstone`:: The SMT behaves as it does when you select the `rewrite` option, except that it also retains `TOMBSTONE` records.
```

Comment (roldanbob) [reply documentation/modules/ROOT/pages/transformations/event-flattening.adoc:291]: Moved the note to the follow immediately after the first paragraph of the description.

File: documentation/modules/ROOT/pages/transformations/mongodb-event-flattening.adoc [hunk 2]
```diff
@@ -489,6 +489,10 @@ For more information about the `array.coding` option, see the xref:mongodb-event
 |{prodname} generates a tombstone record for each `delete` operation.
 The default behavior is that event flattening SMT removes tombstone records from the stream.
 To retain tombstone records in the stream, specify `drop.tombstones=false`.
+[NOTE]
+====
+Note: It will be removed in further release, use xref:mongodb-extract-new-record-state-delete-tombstone-handling-mode[`delete.tombstone.handling.mode`] instead
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/mongodb-event-flattening.adoc:494]:
```suggestion
This option is scheduled for removal in a future release.
In its place, use the xref:mongodb-extract-new-record-state-delete-tombstone-handling-mode[`delete.tombstone.handling.mode`] option.
```

File: documentation/modules/ROOT/pages/transformations/mongodb-event-flattening.adoc [hunk 3]
```diff
@@ -503,6 +507,23 @@ To provide another way to indicate that the record was deleted, the modified rec
  +
 If you set the `rewrite` option, you might find that the updated, simplified records for `DELETE` operations are sufficient for tracking deleted records.
 In such a case, you might want the SMT to xref:mongodb-extract-new-record-state-drop-tombstones[drop tombstone records].
+[NOTE]
+====
+Note: It will be removed in further release, use xref:mongodb-extract-new-record-state-delete-tombstone-handling-mode[`delete.tombstone.handling.mode`] instead
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/mongodb-event-flattening.adoc:512]:
```suggestion
This option is scheduled for removal in a future release.
In its place, use the xref:mongodb-extract-new-record-state-delete-tombstone-handling-mode[`delete.tombstone.handling.mode`] option.
```

File: documentation/modules/ROOT/pages/transformations/event-flattening.adoc [hunk 4]
```diff
@@ -268,6 +271,24 @@ Specify `none` to keep the change event record in the stream. The record contain
 Specify `rewrite` to keep the change event record in the stream and edit the record to have a `value` field that contains the key/value pairs that were in the `before` field and also add `+__deleted: true+` to the `value`. This is another way to indicate that the record has been deleted. +
  +
 When you  specify `rewrite`, the updated simplified records for `DELETE` operations might be all you need to track deleted records. You can consider accepting the default behavior of dropping the tombstone records that the {prodname} connector creates.
+[NOTE]
+====
+Note: It will be removed in further release, use xref:extract-new-record-state-delete-tombstone-handling-mode[`delete.tombstone.handling.mode`] instead
+====
+
+|[[extract-new-record-state-delete-tombstone-handling-mode]]xref:extract-new-record-state-delete-tombstone-handling-mode[`delete.tombstone.handling.mode`]
+|
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/event-flattening.adoc:280]:
```suggestion
|No default
```

File: documentation/modules/ROOT/pages/transformations/mongodb-event-flattening.adoc [hunk 5]
```diff
@@ -503,6 +507,23 @@ To provide another way to indicate that the record was deleted, the modified rec
...
+|{prodname} generates a change event record for each `DELETE` operation. It will determine the event flattening SMT how to handle `DELETE` events from the stream.
+
+`drop`:: The SMT removes both the `DELETE` event and `TOMBSTONE` from the stream.
+`tombstone`:: The SMT only keeps the `TOMBSTONE` in the stream. The record contains only `"value": "null"`.
+`rewrite`:: The SMT keeps the change event record in the stream and edit the record to have a `value` field that contains the key/value pairs that were in the `before` field and also add `+__deleted: true+` to the `value` and remove the `TOMBSTONE`. This is another way to indicate that the record has been deleted.
+`rewrite-with-tombstone`:: The SMT has the same behavior from `rewrite` except to keep the `TOMBSTONE`.
+[NOTE]
+====
+Note: When specify a value from above will enable this new config in event flatting SMT, it has a higher priority than the deprecated xref:mongodb-extract-new-record-state-drop-tombstones[`drop.tombstones`] and xref:mongodb-extract-new-record-state-delete-handling-mode[`delete.handling.mode`].
+====
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/mongodb-event-flattening.adoc:526]:
```suggestion
|[[mongodb-extract-new-record-state-delete-tombstone-handling-mode]]xref:mongodb-extract-new-record-state-delete-tombstone-handling-mode[`delete.tombstone.handling.mode`]
|No default
|{prodname} generates a change event record for each `DELETE` operation. 
This setting determines how the MongoDB event flattening SMT handles `DELETE` events in the stream.

[NOTE]
====
The setting for this option takes precedence over any conflicting settings that you might configure for the deprecated xref:mongodb-extract-new-record-state-drop-tombstones[`drop.tombstones`] or xref:mongodb-extract-new-record-state-delete-handling-mode[`delete.handling.mode`] options.
====

Set one of the following options:

`drop`:: The SMT removes both `DELETE` events and `TOMBSTONE`records from the stream.
`tombstone` (default):: The SMT retains `TOMBSTONE` records in the stream.
The `TOMBSTONE` record contains only the following value: `"value": "null"`.
`rewrite`:: The SMT retains the change event record in the stream and makes the following changes:

* Adds a `value` field to the record that contains the key/value pairs from the `before` field of the original record.
* Adds `+__deleted: true+` to the `value` of the record.
* Removes `TOMBSTONE` records.
+
This setting provides another way to indicate that the record has been deleted.

`rewrite-with-tombstone`:: The SMT behaves as it does when you select the `rewrite` options, except that it also retains `TOMBSTONE` records.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]

**Comment by harveyyue** [comment]
@roldanbob Done, a big thank for helping improve the doc description of this option.

**Comment by roldanbob** [comment]
> @roldanbob Done, a big thank for helping improve the doc description of this option.

Oh, you're very welcome, @harveyyue! Thanks for that. I hoped that you would be in agreement with the proposed changes.

**Comment by jpechane** [comment]
@harveyyue @roldanbob Applied, thanks guys!

**Pull request merged**